jaxrs/src/main/java/feign/jaxrs/JAXRSContract.java

```diff
@@ -43,13 +43,19 @@
   static final String ACCEPT = "Accept";
   static final String CONTENT_TYPE = "Content-Type";
 
+  // Protected so unittest can call us
+  // XXX: Should parseAndValidateMetadata(Class, Method) be public instead? The old deprecated parseAndValidateMetadata(Method) was public..
```

Comment:
public might not hurt, but protected is probably the right relationship. I think your first line covers why. If/when we refactor the test, we can undo this easier than undoing publicing something.